dataset_loaders/images/isbi_em_stacks.py

```diff
@@ -52,20 +58,52 @@ class IsbiEmStacksDataset(ThreadedDataset):
         1: (255, 255, 255)}  # Membranes
     _mask_labels = {0: 'Non-membranes', 1: 'Membranes'}
 
-    def __init__(self, which_set='train', split=0.85, *args, **kwargs):
+    def __init__(self, which_set='train', split=0.60, crossval=5, fold=3, rand_perm=None,
+                 *args, **kwargs):
 
         assert which_set in ["train", "valid", "val", "test"]
         self.which_set = "val" if which_set == "valid" else which_set
 
-        if self.which_set == "train":
-            self.start = 0
-            self.end = int(split*30)
-        elif self.which_set == "val":
-            self.start = int(split*30)
-            self.end = 30
-        elif self.which_set == "test":
-            self.start = 0
-            self.end = 30
+        self.middle_fold = False  # False by default
+        if crossval is not None:  # if cross-validation is used
+            if rand_perm is not None:
+                self.rand_indexes=rand_perm
+            else:
+                self.rand_indexes=range(0,30)
+            # number of images per fold
+            img_per_fold = int(30/crossval)
+            # start and end index for validation fold
+            start = (fold-1)*img_per_fold
+            end = fold*img_per_fold
+
+            if self.which_set == "train":
+                if fold == 1:  # if it is the first fold
+                    self.start = end
+                    self.end = 30
+                elif fold == crossval:  # if it is the last fold
+                    self.start = 0
+                    self.end = start
+                else:  # a fold in the middle of the dataset
+                    self.middle_fold = True
+                    self.start = start
+                    self.end = end
+            elif self.which_set == "val":
+                self.start = start
+                self.end = end
+            elif self.which_set == "test":
```

Comment:
More generally, I suggest to replace L75-93 with the following approach:
```python
if self.which_set == "train":
    self.start_1 = 0
    self.end_1 = fold * img_per_fold
    self.start_2 = (fold+1) * img_per_fold
    self.end_2 = 30
elif self.which_set == "val":
    self.start_1 = fold * img_per_fold
    self.end_1 = self.start_2 = self.end_2 = (fold+1) * img_per_fold
elif self.which_set == "test":
    raise ValueError('Cannot perform cross-validation on test.')
```
and then replace self.get_names with
```python
return {'default': self.rand_indexes[self.start_1:self.end_1] + 
                   self.rand_indexes[self.start_2:self.end_2]}
```